### PR TITLE
VarInt: add reading from and writing into `ByteBuffer`

### PR DESCRIPTION
**Pull request by schildbach**
This is a followup PR to #2845.

### CODE REVIEW

**Comment by schildbach** [comment]
Alternatively the methods to read and write the `VarInt` could be put into `ByteUtils` next to the other read and write helpers.

**Review by msgilligan — APPROVED** [review]
LGTM

File: core/src/main/java/org/bitcoinj/core/AddressMessage.java [hunk 1]
```diff
@@ -37,7 +37,7 @@ public abstract class AddressMessage extends Message {
     protected void bitcoinSerializeToStream(OutputStream stream) throws IOException {
         if (addresses == null)
             return;
-        stream.write(new VarInt(addresses.size()).encode());
+        stream.write(VarInt.of(addresses.size()).encode());
```
[the remark this answers is not in the document]
Comment (msgilligan) [reply core/src/main/java/org/bitcoinj/core/AddressMessage.java:40]: This is both out-of-scope and a comment on the first of the 2 commits currently in this PR.

**Comment by schildbach** [comment]
Merged.

**Pull request closed without merging**